src/java/org/jivesoftware/openfire/session/LocalSession.java

```diff
@@ -413,4 +429,27 @@ public String getCipherSuiteName() {
         }
         return "NONE";
     }
+
+    /**
+     * Enables stream management for session
+     * @param enable XEP-0198 <enable/> stanza
+     */
+    public void enableStreamMangement(Element enable) {
+
+    	// Do nothing if already enabled
+    	if(streamManager.isEnabled()) {
+    		return;

```

Comment:
Appears that <enable> while already enabled doesn't seem to be covered in xep198. I think the spec should state that sending a <enable> after the SM has already been enabled is either a no-op (like it's implemented here), or result in a stream-error and subsequent stream termination (which I would prefer).